### ISSUE TITLE
Cannot update password for user

### ISSUE THREAD

**Issue by Gregoryr-ng** [jgrodziski/keycloak-clojure#19]
I am not using admin client in master realm. I am using a newly created realm and client and I have assigned all the admin permission to the client.
I tried to call user/update-user! and passed email, firstname, lastname, pasword to see how it works,  everything updated except for password. But my actual intention is to update the password. Do you have any idea? thanks

**Comment by jgrodziski** [comment]
Yes, this is the current behaviour as IMHO the password change feature should be used with caution and should be made by the user with the Keycloak update/lost password feature.
I push a refactoring commit, but without proper tests for the moment as I didn't have the time right now to write them. Could you tell me if it works nonetheless?

**Comment by jgrodziski** [comment]
I didn't publish a new version as there isn't a proper test for the moment but you could include the last version with tools.deps and `{:git/url "https://github.com/jgrodziski/keycloak-clojure" :sha "ba5488b9eee3ff6c67311615ce445b3c92c680af"}`

**Comment by Gregoryr-ng** [comment]
I see. I just googled a bit for that, it is considered unsafe to support credentials modifications via REST API. They now only supports AIA for user, or Admin Rest API for backend to handle user reset password request. So, I should use Admin Rest API to handle reset password for user right? If this is the case, for this part, I think I should directly call keycloak /reset-password API instead?

Appreciate your work!! thanks

**Comment by iusshpandeh** [comment]
Are you willing to share snippets on how did you integrate the keycloak-clojure in your project? I've been trying to integrate it to my Ring API and have failed miserably. Any help would be appreciated.

**Comment by jgrodziski** [comment]
I have some Ring middlewares regarding authentication, access control headers and authorization but not very battle tested for the moment. I can nonetheless merge them in Keycloak-Clojure asap.

**Comment by iusshpandeh** [comment]
> 
> 
> I have some Ring middlewares regarding authentication, access control headers and authorization but not very battle tested for the moment. I can nonetheless merge them in Keycloak-Clojure asap.

How long might it take? Its been quite a long already. Thanks.

**Comment by jgrodziski** [comment]
Yes it is on its way, my clients work takes me lot of time at the moment. I can nevertheless commit the ring middlewares and let you know.
